File: backend/app/schemas/booking.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.booking import BookingStatus
from app.schemas.common import ContactInfo


class BookingCreate(BaseModel):
    lot_id: int
    contact: ContactInfo


class BookingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    lot_id: int
    contact_name: str
    contact_info: str
    status: BookingStatus
    created_at: datetime

File: frontend/src/types/booking.ts
import type { ContactInfo } from './common'

export type BookingStatus = 'active' | 'cancelled'

export interface BookingCreate {
  contact: ContactInfo
}

export interface Booking {
  id: number
  lot_id: number
  contact_name: string
  contact_info: string
  status: BookingStatus
  created_at: string
}
